File: apps/cargo-tracker/src/main/java/com/example/cargotracker/booking/domain/model/valueobjects/BookingId.java
package com.example.cargotracker.booking.domain.model.valueobjects;

import java.util.UUID;

/**
 * 予約識別子。
 *
 * @param value UUID
 */
public record BookingId(UUID value) {

    public BookingId {
        if (value == null) {
            throw new IllegalArgumentException("予約 ID は必須です");
        }
    }

    public static BookingId generate() {
        return new BookingId(UUID.randomUUID());
    }

    public static BookingId of(String value) {
        return new BookingId(UUID.fromString(value));
    }
}

File: docs/article/source/csharp/apps/cargo-tracker/src/CargoTracker.Web/Booking/Domain/Model/BookingId.cs
namespace CargoTracker.Booking.Domain.Model;

/// <summary>予約識別子（業務キー）。</summary>
public sealed record BookingId(string Value)
{
    public static BookingId Generate() => new($"BKG-{DateTimeOffset.UtcNow:yyyyMMdd}-{Guid.NewGuid():N}"[..20].ToUpperInvariant());
}
